test/contractRelayer/relayerProposal.js

```diff
@@ -0,0 +1,131 @@
+
+    it('should revert because proposedRelayerAddress is not a relayer', async () => {
+        // 0 == RelayerActionType.Remove
+        await TruffleAssert.reverts(
+            RelayerInstance.voteRelayerProposal(proposedRelayerAddress, 0),
+            'proposed address is not a relayer');
+    });
+
+    it('should revert because proposedRelayerAddress is already a relayer', async () => {
+        // 1 == RelayerActionType.Add
+        await TruffleAssert.reverts(
+            RelayerInstance.voteRelayerProposal(relayer2, 1, { from: relayer3 }),
+            'proposed address is already a relayer');
+    });
+
+    it('voteRelayerProposal should emit RelayerProposalCreated event', async () => {
+        const proposalVoteTx = await RelayerInstance.voteRelayerProposal(proposedRelayerAddress, 1);
+        TruffleAssert.eventEmitted(proposalVoteTx, 'RelayerProposalCreated', event => {
+            return event.proposedAddress === proposedRelayerAddress &&
+                event.relayerActionType.toNumber() === 1
+        });
+    });
+
+    it('voteRelayerProposal should emit RelayerProposalVote event', async () => {
+        // 1 == RelayerActionType.Add
+        const proposalVoteTx = await RelayerInstance.voteRelayerProposal(proposedRelayerAddress, 1);
+        TruffleAssert.eventEmitted(proposalVoteTx, 'RelayerProposalVote', event => {
+            return event.proposedAddress === proposedRelayerAddress &&
+                event.vote.toNumber() === 1 // 1 == VoteStatus.Active
+        });
+    });
+
+    it('voteRelayerProposal should emit RelayerAdded event', async () => {
+        // 1 == RelayerActionType.Add
+        await TruffleAssert.passes(RelayerInstance.voteRelayerProposal(proposedRelayerAddress, 1));
+        const proposalVoteTx = await RelayerInstance.voteRelayerProposal(proposedRelayerAddress, 1, { from: relayer2 });
+        TruffleAssert.eventEmitted(proposalVoteTx, 'RelayerAdded', event => {
+            return event.relayerAddress === proposedRelayerAddress
+        });
+    });
+
+    it('voteRelayerProposal should emit RelayerRemoved event', async () => {
+        // 1 == RelayerActionType.Add
+        await TruffleAssert.passes(RelayerInstance.voteRelayerProposal(relayer2, 0));
+        const proposalVoteTx = await RelayerInstance.voteRelayerProposal(relayer2, 0, { from: relayer3 });
+        TruffleAssert.eventEmitted(proposalVoteTx, 'RelayerRemoved', event => {
+            return event.relayerAddress === relayer2
+        });
+    });
+
+    it("should set relayer proposal' status to inactive, because relayerProposal._yesVotes >= _relayerThreshold", async () => {
+        const expectedRelayerProposalValues = [
+            1,                    // _action, 1 == RelayerActionType.Add
+            [relayer1, relayer2], // _yesVotes
+            [],                   // _noVotes
+            0                     // _status, 0 == VoteStatus.Inactive
+        ];
+        
+        // 1 == RelayerActionType.Add
```

Comment:
I can't seem to find where `RelayerActionType` is declared?

Reply:
Yea, it also took me a second to realize that `Relayer.sol` has an interface, but `RelayerActionType` is declared in `IRelayer.sol` [here](https://github.com/ChainSafe/chainbridge-solidity/blob/3bb85bb88e99dd7a0adfcf677cf1f236a00723f4/contracts/interfaces/IRelayer.sol#L5)